es6-shim.js

```diff
@@ -756,6 +771,64 @@
   });
   addIterator(ArrayIterator.prototype);
 
+  var ObjectIterator = function (object, kind) {
+    this.object = object;
+    this.i = 0;
+    // Don't generate keys yet.
+    this.array = null;
+    this.kind = kind;
+  };
+
+  defineProperties(ObjectIterator.prototype, {
+    next: function () {
+      var key, array = this.array;
+      if (!(this instanceof ObjectIterator)) {
+        throw new TypeError('Not an ObjectIterator');
+      }
+
+      if (typeof array !== 'undefined') {
+        // Keys not generated
+        if (array === null) {
+          array = this.array = [];
+
+          for (key in this.object) {
+            array.push(key);
+          }
+        }
+
+        var len = ES.ToLength(array.length);
+
+        // Find current index, then update i
+        while (this.i < len) {
+          key = array[this.i];
+          this.i++;
+
+          // The candidate key isn't defined on object.
+          // Must have been deleted, or object[[Prototype]]
+          // has been modified.
+          if (!(key in this.object)) {
+            continue;
+          }
+
+          switch (this.kind) {
```

Comment:
A good rule: only use switch when a program _must_ take advantage of multiple case fall-through, otherwise use if/else (as in this example).